3496 - Geo UI Refactor - Statistics UI

yaguzmang opened the pull request:
Adding the statistics sections to the side panel, reducing the padding of the cells in the tables, and deleting the old statistics UI components.

https://github.com/openforis/fra-platform/assets/41337901/a55dbd07-2816-4fab-bb67-72aeee01cc9a


To do: 
- Add scrolling for NavGeo.
- Add styles and translation to the error message if the map fails to load.

yaguzmang committed:
3496 - Add option to export custom csv data with ButtonTableExport

yaguzmang committed:
3496 - Update StatisticsTable & delete ButtonStatisticsTableExport

yaguzmang committed:
3496 - Move components to StatisticsSidePanel

yaguzmang committed:
3496 - Move StatisticsTable to StatisticsSidePanel

yaguzmang committed:
3496 - Update table titles style

yaguzmang committed:
3496 - Add StatisticsSection component & render all sections

yaguzmang committed:
3496 - Delete old geo statistics code

yaguzmang commented on src/client/pages/Geo/GeoMapWrapper/geoMapWrapper.scss:
```diff
@@ -1,33 +1 @@
 @import 'src/client/style/partials';
```
This file wasn't deleted because I will use it in an upcoming PR to add styles to the error message when the map fails to load. I tried deleting it and renaming it with the capital letter -> GeoMapWrapper.scss, but this would cause the same issue we had before with "Fileusages.scss".

minotogna commented on src/client/components/ButtonTableExport/ButtonTableExport.tsx:
```diff
@@ -25,6 +29,21 @@ const ButtonTableExport: React.FC<Props> = (props) => {
 
   if (print) return null
 
+  if (!Objects.isEmpty(csvData)) {
```
Is this needed for geo part ? Can u use a grid instead and use the button grid export ?
If not , let's add a custom csv button export only for geo and under geo .
We'll need to remove those tables in the future .

yaguzmang replied:
Yes, we need it for geo. There is one table where we need to export a csv file that's slightly different from what's on the screen:
![image](https://github.com/openforis/fra-platform/assets/41337901/1a865c9c-2760-4e2c-88ba-4965a83a8c6f)

The exported csv has an extra column "Land area", and the unit of Forest area is only in the header:
![image](https://github.com/openforis/fra-platform/assets/41337901/48e0d742-1738-41a7-8c84-0293e94cfe8c)

yaguzmang replied:
Added a button under geo to export the custom CSV, and updated the table component to use DataGrid instead. The video is now updated 👍

minotogna replied:
Nice , thank you !

yaguzmang committed:
3496 - Revert changes to ButtonTableExport & add ButtonCSVExport

yaguzmang committed:
3496 - Use datagrid instead of html table

minotogna commented:
Can the csv download button be on top-left corner of the table ? Now i see it in the bottom right corner

minotogna commented on src/client/pages/Geo/ButtonCSVExport/ButtonCSVExport.tsx:
```diff
@@ -0,0 +1,34 @@
+
+  return (
+    <CSVLink
+      className="data-grid__btn-export btn-xs btn-primary no-print"
```
Use the useButtonClassName hook . See other csv download button (for example in data table )

yaguzmang replied:
Done. I added it and noticed that the style didn't match the ButtonGridExport style, so I checked ButtonGridExport and realized it did not match the style of ButtonTableExport.

For example here, ButtonGridExport on the top, ButtonTableExport on the bottom.
![Screenshot 2024-03-15 at 11 17 04 AM](https://github.com/openforis/fra-platform/assets/41337901/9488dac8-ce98-4d14-adfb-0e33155f2d9b)

I updated ButtonGridExport to use the same logic and styles introduced to ButtonTableExport in the `Files and links per Cycle` feature:

![image](https://github.com/openforis/fra-platform/assets/41337901/b78dad84-ed3b-4b23-be77-58b32dd1d8a6)

And now the export button for Geo matches those previous two (on the top left now):
![image](https://github.com/openforis/fra-platform/assets/41337901/1a56709f-6c83-4bef-a748-cac20b5a107b)

yaguzmang committed:
3496 - Update buttons style

Updating the styles of ButtonGridExport so it matches
the style of ButtonTableExport (recently updated). And,
updating the styles of ButtonCSVExport so it matches those
previous two.

yaguzmang committed:
3496 - Open side panel by default

yaguzmang committed:
3496 - Move export button to top left

yaguzmang committed:
3496 - Keep previous disabled property logic in ButtonGridExport

mergify[bot] committed:
Merge branch '3496-geo-ui-refactor' into 3496-geo-ui-refactor-statistics-ui
